src/EnergyPlus/CurveManager.cc

```diff
@@ -3267,64 +3180,43 @@ namespace CurveManager {
-        Term2 = 6.9 / ReynoldsNumber;
-        Term3 = -1.8 * std::log10(Term1 + Term2);
+        Real64 const Term1 = std::pow(RoughnessRatio / 3.7, 1.11);
+        Real64 const Term2 = 6.9 / ReynoldsNumber;
+        Real64 const Term3 = -1.8 * std::log10(Term1 + Term2);
         if (Term3 != 0.0) {
-            CalculateMoodyFrictionFactor = std::pow(Term3, -2.0);
+            return std::pow(Term3, -2.0);
         } else {
             if (!state.dataCurveManager->FrictionFactorErrorHasOccurred) {
-                RR = format("{:.7R}", RoughnessRatio);
-                Re = format("{:.1R}", ReynoldsNumber);
                 ShowSevereError(state, "Plant Pressure System: Error in moody friction factor calculation");
-                ShowContinueError(state, "Current Conditions: Roughness Ratio=" + RR + "; Reynolds Number=" + Re);
+                ShowContinueError(state,
+                                  format("Current Conditions: Roughness Ratio={:.7R}; Reynolds Number={:.1R}", RoughnessRatio, ReynoldsNumber));
                 ShowContinueError(state, "These conditions resulted in an unhandled numeric issue.");
                 ShowContinueError(state, "Please contact EnergyPlus support/development team to raise an alert about this issue");
                 ShowContinueError(state, "This issue will occur only one time.  The friction factor has been reset to 0.04 for calculations");
                 state.dataCurveManager->FrictionFactorErrorHasOccurred = true;
             }
-            CalculateMoodyFrictionFactor = 0.04;
+            return 0.04;
         }
-
-        return CalculateMoodyFrictionFactor;
     }
 
     void checkCurveIsNormalizedToOne(EnergyPlusData &state,
-                                     std::string const callingRoutineObj, // calling routine with object type
-                                     std::string const objectName,        // parent object where curve is used
-                                     int const curveIndex,                // index to curve object
-                                     std::string const cFieldName,        // object field name
-                                     std::string const cFieldValue,       // user input curve name
-                                     Real64 const Var1,                   // required 1st independent variable
-                                     Optional<Real64 const> Var2,         // 2nd independent variable
-                                     Optional<Real64 const> Var3,         // 3rd independent variable
-                                     Optional<Real64 const> Var4,         // 4th independent variable
-                                     Optional<Real64 const> Var5,         // 5th independent variable
-                                     Optional<Real64 const> Var6          // 6th independent variable
+                                     std::string const &callingRoutineObj, // calling routine with object type
```

Comment:
The order of fields here is killing me. Same comment about passing a bunch of error strings to a function on the rare chance that there is an error.

Reply:
:+1: